### PR TITLE
rdf-js: Use single generic for I/O types of datafactory functions

### PR DESCRIPTION
**Pull request by LaurensRietveld**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29836#issuecomment-434702390
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by rubensworks — APPROVED** [review]

File: types/rdf-js/index.d.ts [hunk 1]
```diff
@@ -297,7 +297,7 @@ export interface DataFactory {
      * @see Triple
      * @see DefaultGraph
      */
-    triple<Q_In extends BaseQuad = Quad,  Q_Out extends BaseQuad = Quad>(subject: Q_In['subject'], predicate: Q_In['predicate'], object: Q_In['object']): Q_Out;
+    triple<Q extends BaseQuad = Quad>(subject: Q['subject'], predicate: Q['predicate'], object: Q['object']): Q;
```
Comment (DanielRosenwasser) [inline types/rdf-js/index.d.ts]: Can you check whether we have noUnusedGenerics disabled in the `tslint.json` for this package? If you care to work on them, there may be others in this package.

Comment (DanielRosenwasser) [reply types/rdf-js/index.d.ts]: Yes, if you remove that line entirely it should enable that rule.

Comment (rubensworks) [reply types/rdf-js/index.d.ts]: @LaurensRietveld Any update on this one?
I'd be happy to update it myself if you can give me access to your fork.

Comment (LaurensRietveld) [reply types/rdf-js/index.d.ts]: I'm a bit confused ;). What is the problem and what is the proposed solution?

Is the problem having the `no-unnecessary-generics` flag set to false? If so, what is the recommended solution?
I think this situation results in a false-positive for this lint rule. If I see the doc for that rule, I see that it `Forbids a function to use a generic type parameter only once`. We're not using it once right? We're using if for 3 arguments and the return type.

Comment (rubensworks) [reply types/rdf-js/index.d.ts]: > If so, what is the recommended solution?

I think @DanielRosenwasser just means that `no-unnecessary-generics` should be set to true by removing the line from `tslint.json`.

> We're not using it once right? We're using if for 3 arguments and the return type.

Indeed, it should be fine here. But I guess there's nothing wrong in double-checking and making the compiler settings a bit more restrictive :-)

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**